clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/restapiv2/requests/SetNodeStateRequest.java

```diff
@@ -95,15 +97,92 @@ static SetResponse setWantedState(
                 " wanted-state=" + wantedState +
                 " new-wanted-state=" + newWantedState +
                 " change-check=" + result);
+
+        boolean success = ensureWantedState(
+                result,
+                newWantedState,
+                condition,
+                nodeInfo,
+                cluster,
+                stateListener);
+
+        // If the state was successfully set, just return an "ok" message back.
+        String reason = success ? "ok" : result.getReason();
+        return new SetResponse(reason, success);
+    }
+
+    /**
+     * Returns true if the current/old wanted state already matches the requested
+     * wanted state, or the requested state has been accepted as the new wanted state.
+     */
+    private static boolean ensureWantedState(NodeStateChangeChecker.Result result,
```

Comment:
I feel that the name of this method leaves me a bit hanging with regards to what the method's semantics are. "ensure wanted state.. is/does what?"

Reply:
Changed to 'setWantedStateAccordingToResult'